app/javascript/mastodon/features/ui/components/media_modal.tsx

```diff
@@ -0,0 +1,359 @@
+import { forwardRef, useCallback, useEffect, useMemo, useState } from 'react';
+import type { RefCallback, FC } from 'react';
+
+import { defineMessages, useIntl } from 'react-intl';
+
+import classNames from 'classnames';
+
+import type { List as ImmutableList } from 'immutable';
+
+import { animated, useSpring } from '@react-spring/web';
+import { useDrag } from '@use-gesture/react';
+
+import type { MediaAttachment } from '@/mastodon/models/status';
+import ChevronLeftIcon from '@/material-icons/400-24px/chevron_left.svg?react';
+import ChevronRightIcon from '@/material-icons/400-24px/chevron_right.svg?react';
+import CloseIcon from '@/material-icons/400-24px/close.svg?react';
+import FitScreenIcon from '@/material-icons/400-24px/fit_screen.svg?react';
+import ActualSizeIcon from '@/svg-icons/actual_size.svg?react';
+import type { RGB } from 'mastodon/blurhash';
+import { getAverageFromBlurhash } from 'mastodon/blurhash';
+import { GIFV } from 'mastodon/components/gifv';
+import { Icon } from 'mastodon/components/icon';
+import { IconButton } from 'mastodon/components/icon_button';
+import { Footer } from 'mastodon/features/picture_in_picture/components/footer';
+import { Video } from 'mastodon/features/video';
+
+import { ZoomableImage } from './zoomable_image';
+
+const messages = defineMessages({
+  close: { id: 'lightbox.close', defaultMessage: 'Close' },
+  previous: { id: 'lightbox.previous', defaultMessage: 'Previous' },
+  next: { id: 'lightbox.next', defaultMessage: 'Next' },
+  zoomIn: { id: 'lightbox.zoom_in', defaultMessage: 'Zoom to actual size' },
+  zoomOut: { id: 'lightbox.zoom_out', defaultMessage: 'Zoom to fit' },
+});
+
+interface MediaModalProps {
+  media: ImmutableList<MediaAttachment>;
+  statusId?: string;
+  lang?: string;
+  index: number;
+  onClose: () => void;
+  onChangeBackgroundColor: (color: RGB | null) => void;
+  currentTime?: number;
+  autoPlay?: boolean;
+  volume?: number;
+}
+
+export const MediaModal: FC<MediaModalProps> = forwardRef<
+  HTMLDivElement,
+  MediaModalProps
+>(
+  (
+    {
+      media,
+      onClose,
+      index: startIndex,
+      lang,
+      currentTime,
+      autoPlay,
+      volume,
+      statusId,
+      onChangeBackgroundColor,
+    },
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- _ref is required to keep the ref forwarding working
+    _ref,
```

Comment:
But it's not being used/forwarded? You'd have to assign to the ref inside of `handleRef`

Reply:
no it's not being used. unfortunately if you create a modal that doesn't use `forwardRef` it causes React to throw warnings for reason i haven't investigated, but it's not actually used anywhere as modals are exclusively opened by `ModalRoot`

Reply:
Oof, I see now. This really needs to be cleaned up at some point. 🙈 It doesn't seem like the `components/modal_root` component is ever assigned the ref prop, and the `features/.../modal_root` component only seems to require a ref for one specific use case that's deprecated now (using ref to get a class component's instance… [EDIT: Actually, that's been refactored to use an imperative handle]), and that's what's causing these console warnings.